common/views/components/NewsletterSignup/NewsletterSignup.js

```diff
@@ -0,0 +1,193 @@
+          <form
+            noValidate={this.state.noValidate}
+            onSubmit={this.handleSubmit}
+            name='newsletter-signup'
+            id='newsletter-signup'
+            action='https://r1-t.trackedlink.net/signup.ashx'
+            method='post'>
+            <input type='hidden' name='userid' value='225683' />
+            <input type='hidden' name='ReturnURL' value='http://localhost:3000/info/newsletter' />
+            <input type='hidden' name='SIG22a9ece3ebe9b2e10e328f234fd10b3f5686b9f4d45f628f08852417032dc990' value='' />
+
+            <fieldset className={spacing({s: 2}, {margin: ['bottom']})}>
+              <legend>What are you interested in? Choose as many as you like:</legend>
+              <ul className='plain-list no-padding'>
+                {addressBooks.map((addressBook) => (
+                  <li className={spacing({s: 2}, {margin: ['bottom']})} key={addressBook.id}>
+                    <HTMLInput
+                      id={addressBook.id}
+                      type='checkbox'
+                      name={addressBook.name}
+                      label={addressBook.label}
+                      onChange={this.updateCheckedInputs} />
+                    <p className={`${font({s: 'HNL6'})} ${spacing({s: 1}, {margin: ['top']})}`}>{addressBook.description}</p>
+                  </li>
+                ))}
+              </ul>
+            </fieldset>
+
+            <div className={spacing({s: 2}, {margin: ['bottom']})}>
+              <HTMLInput
+                required={true}
+                id='email'
+                type='email'
+                name='Email'
+                label='Your email address'
+                placeholder='Your email address'
+                isLabelHidden={true}
+                onChange={this.handleEmailInput}
+              />
+            </div>
+
+            <p className={font({s: 'HNL6'})}>We use a third party provider, Dotmailer, to deliver our newsletters. For information about how we handle your data, please read our <a href='https://wellcome.ac.uk/about-us/privacy-and-terms'>privacy notice</a>. You can unsubscribe at any time using links in the emails you receive.</p>
+
+            <Button
+              extraClasses={`btn--primary ${spacing({s: 2}, {margin: ['bottom']})}`}
+              text='Submit' />
+
+            {this.state.isCheckboxError && this.state.isSubmitAttempted &&
+              <p className={`${spacing({s: 2}, {padding: ['top', 'right', 'bottom', 'left'], margin: ['bottom']})} border-width-1 border-color-red font-red`}>Please select an option.</p>
+            }
+
+            {this.state.isEmailError && this.state.isSubmitAttempted &&
+              <p className={`${spacing({s: 2}, {padding: ['top', 'right', 'bottom', 'left'], margin: ['bottom']})} border-width-1 border-color-red font-red`}>Please enter a valid email address.</p>
+            }
+          </form>
+        }
+      </div>
+    );
+  }
+}
+
+export default NewsletterSignup;
```

Comment:
I'm a little confused as to why we have both the React client handling as well as the `client/app.js` handling?
It makes it quite confusing as to what is what, and it seems like a lot of testing code has made it in?

Reply:
We'll need the component to work once we start using it in React, so felt like it made sense to write equivalent validation code for the React and non-React versions at the same time.